samcli/lib/sync/flows/zip_function_sync_flow.py

```diff
@@ -70,10 +69,19 @@ def __init__(
         self._s3_client = None
         self._artifact_folder = None
         self._zip_file = None
-        self._local_sha = None
         self._build_graph = None
         self._color = Colored()
 
+    @property
+    def sync_state_identifier(self) -> str:
+        """
+        Sync state is the unique identifier for each sync flow
+        In sync state toml file we will store
+        Key as ZipFunctionSyncFlow:FunctionLogicalId
+        Value as function ZIP hash
+        """
+        return self.__class__.__name__ + ":" + self._function_identifier
```

Comment:
This method can be moved to `FunctionSyncFlow` class which should have the `self._function_identifier` field.